dialogy/types/entity/time_interval/__init__.py

```diff
@@ -201,9 +201,10 @@ def pick_value(
                     }
                 )
             elif to_:
-                datetime_val[const.FROM][const.VALUE] = (
+                from_or_to = const.FROM if from_datetime_val else const.TO
+                datetime_val[from_or_to][const.VALUE] = (
                     datetime.fromisoformat(
-                        datetime_val.get(const.FROM, {}).get(const.VALUE)
+                        datetime_val.get(const.TO, {}).get(const.VALUE)
```

Comment:
@biswaroop1547 This will also operate in a case where only TO was present and was inside constraints. In that case we'll replace the value of TO to the lower bound of time constraint. Shouldn't it be set to upper bound of time constraint?

Reply:
Yes you're right. Updated, please check once 🙏🏻